Library/ViewModels/PledgePaymentMethodsViewModel.swift

```diff
@@ -283,28 +296,55 @@ public final class PledgePaymentMethodsViewModel: PledgePaymentMethodsViewModelT
 
     let createSetupIntentEvent = Signal.combineLatest(
       project,
-      paymentSheetOnPledgeContext.filter(isTrue)
+      pledgeTotal,
+      paymentSheetType
     )
     .takeWhen(didTapToAddNewCard)
-    .switchMap { (project, _) -> SignalProducer<Signal<PaymentSheetSetupData, ErrorEnvelope>.Event, Never> in
-      AppEnvironment.current.apiService
+    .switchMap { (project, pledgeTotal, paymentSheetType) -> SignalProducer<
+      Signal<PaymentSheetSetupData, ErrorEnvelope>.Event,
+      Never
+    > in
+
+    let clientSecretSignal: SignalProducer<String, ErrorEnvelope>
+
+    switch paymentSheetType {
+    case .setupIntent:
+      clientSecretSignal = AppEnvironment.current.apiService
         .createStripeSetupIntent(input: CreateSetupIntentInput(projectId: project.graphID))
-        .ksr_debounce(.seconds(1), on: AppEnvironment.current.scheduler)
-        .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
-        .switchMap { envelope -> SignalProducer<PaymentSheetSetupData, ErrorEnvelope> in
+        .map { $0.clientSecret }
 
-          var configuration = PaymentSheet.Configuration()
-          configuration.merchantDisplayName = Strings.general_accessibility_kickstarter()
-          configuration.allowsDelayedPaymentMethods = true
+    case .paymentIntent:
+      assert(
+        !pledgeTotal.isNaN,
```

Comment:
This is my protection for the slightly ugly plumbing, here. I can't think of a good way around passing in the `pledgeTotal` in the overall configuration of this page - there's nowhere else we can get it from, correct?